src/pipeline_edsl/pipeline.ml

```diff
@@ -885,4 +898,21 @@ module Compiler = struct
       seq2hla_hla_types_step ~compiler p
       |> apply_with_metadata ~metadata_spec
 
+  let rec optitype_hla_types_step ~compiler (pipeline : optitype_hla_types pipeline) =
+    let { machine ; work_dir; _ } = compiler in
+    match pipeline with
+    | Optitype (kind, sample) ->
+      begin match sample with
+      | Paired_end_sample (info, l1, l2) ->
+        let r1 = fastq_step ~read:(`R1 info.fragment_id) ~compiler l1 in
+        let r2 = fastq_step ~read:(`R2 info.fragment_id) ~compiler l2 in
+        let work_dir = work_dir // ("optitype_wd_" ^ info.fragment_id) in
```

Comment:
instead of `optitype_wd_` we should use `to_result_prefix` so that running with `` `DNA`` or `` `RNA`` with the same FASTQs does not result in a clash.